### PR TITLE
Allow using a database-level enumeration for primary key

### PR DESCRIPTION
**Pull request by chrisandreae**
Requires PostgreSQL. Combines the advantages of database-level
enumerations with PersistentEnum's ability to define enum constants
and associated data in code, and modify the available constants over
time.

### CODE REVIEW

File: spec/unit/persistent_enum_spec.rb [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+      let(:foreign_key) { foreign_name + "_id" }
+
+      let(:other_model) do
+        fn = foreign_name
+        create_test_model(:referrer, ->(t){ t.integer foreign_key }) do
+          belongs_to_enum fn
+        end
+      end
+
+      it "can be created from enum value" do
```
Comment (thefloweringash) [inline spec/unit/persistent_enum_spec.rb]: These tests used to be close to tests that checked that the `belongs_to_enum` rejected invalid values. I can't find their equivalents.

Comment (chrisandreae) [inline spec/unit/persistent_enum_spec.rb]: added!

File: spec/unit/persistent_enum_spec.rb [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+          acts_as_enum(members)
+        end
+      end
+
+      it_behaves_like "acts like an enum"
+      it_behaves_like "acts like an enum with extra fields"
+    end
+
+    it "must have attributes that match the table" do
+      expect {
```
Comment (thefloweringash) [inline spec/unit/persistent_enum_spec.rb]: Personally I have accepted multi line braced blocks into my heart. It's not exactly normal though. Thoughts?

File: spec/unit/persistent_enum_spec.rb [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+  end
+
+  it "refuses to create a table in a transaction" do
+    expect {
+      ActiveRecord::Base.transaction do
+        create_test_model(:test_create_in_transaction, ->(t){ t.string :name }) do
+          acts_as_enum([:A, :B])
+        end
+      end
+    }.to raise_error(RuntimeError)
```
Comment (thefloweringash) [inline spec/unit/persistent_enum_spec.rb]: `RuntimeError` is very broad, consider matching the text of the error at least.

File: spec/unit/persistent_enum_spec.rb (hunk 2 above)
Comment (chrisandreae) [inline spec/unit/persistent_enum_spec.rb]: Rspec's examples used them. If I had to reverse-engineer a reason why, it's that calling methods on the result of a `do...end` looks odd.

**Pull request merged**